### DISCUSSION TITLE
Troubleshooting DAB config in SWA

### DISCUSSION THREAD

**Discussion by tdutch1** [Azure/data-api-builder#1945] [Q&A]
I have a Static Web App that has been providing APIs from Azure SQL views fine. Today, I introduced a stored procedure API into the config and now all APIs are returning 400 Bad Request. 

If I remove the new "entities" object from the config then everything starts working again. 

I ran DAB locally with the same config and it works fine. 

The question is whether it's possible to get to the logging info that DAB emits when it's running in the SWA. Otherwise, how is possible to troubleshoot config issues in SWA?

Here's my new entry that's breaking things:

    "VehicleProgramsByVinSP": {
      "source": {
        "object": "temp.DabTest_GetVehicleProgramsByVin",
        "type": "stored-procedure"
      },
      "graphql": {
        "enabled": false,
        "type": {
          "singular": "VehicleProgramsByVinSP",
          "plural": "VehicleProgramsByVinSPs"
        }
      },
      "rest": {
        "enabled": true,
        "methods": [
          "get"
        ]
      },
      "permissions": [
        {
          "role": "anonymous",
          "actions": [
            {
              "action": "execute"
            }
          ]
        }
      ]
    }

Thanks for any help or advice on this!

**Comment by tdutch1** [comment]
A related question is how can we determine the DAB version hosted by a Static Web App so that we can make sure we're running the same version locally for development and testing?

**Reply by yorek** [reply]
Very good question :) At the moment the information about DAB is not surfaced via SWA and we're working to figure out a good way to do it. The first step we'll be taking is to make sure that SWA CLI and the SWA hosting in Azure *always* uses the same version so that if you can run your solution locally with the SWA CLI (that behind the scenes runs DAB), you're sure that it will run in SWA too. That should have been the behavior from the beginning, but we noticed that unfortunately that is not always the case. We'll be fixing this soon. A fix should be deployed in our backend in the next days, I'll give you an update here as soon the fix is done.

**Reply by tdutch1** [reply]
Great, thanks for the update.

One way to surface the SWA-hosted DAB version, as well as to help developers compare verbose logging output locally to what's happening in SWA, is to extend Application Insights logging to SWA. Hopefully this is in the roadmap.

Also something that's likely a known issue, but I found that DAB fails to load in the SWA if the telemetry section is added to the config. I'm not sure if the DAB initialization in SWA is throwing an exception with the telemetry section in place or if it can't connect to app insights for some reason. Again, logging/tracing visibility in SWA I think will be essential to reduce the dreaded MS Support tickets.  :)

**Reply by yorek** [reply]
Issue to show version has been created: https://github.com/Azure/data-api-builder/issues/1971. Aside that, latest stable version of DAB (0.9.7) is now being deployed worldwide, so you should be able to have the same experience both when using DAB standalone, and when using DAB via SWA.

**Comment by Koubek** [comment]
Hello everyone,

I’ve been following the discussion and have a related question. I’m trying to understand how DAB is maintained within SWA—specifically, how runtime updates are handled. I couldn’t find detailed documentation addressing this, so any links or further explanations would be greatly appreciated.

We’re considering using DAB, and while DAB within SWA seems to meet our needs, we’re unsure about how DAB runtime updates are managed.

Thanks for your help!
